apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#region
+
+using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Xunit;
+
+#endregion
+
+// ReSharper disable RedundantArgumentDefaultValue
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    public class WriteTests : IClassFixture<TestDatabaseFixture>
+    {
+        public WriteTests(TestDatabaseFixture testFixture)
+        {
+            _testFixture = testFixture;
+        }
+
+        private readonly TestDatabaseFixture _testFixture;
+        private string _lastKey;
+
+        private static string UniqueString() => Guid.NewGuid().ToString();
+
+        private async Task<int> InsertAsync(SpannerParameterCollection values)
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                values.Add("K", _lastKey = UniqueString(), SpannerDbType.String);
+                var cmd = connection.CreateInsertCommand("T", values);
+
+                return await cmd.ExecuteNonQueryAsync();
+            }
+        }
+
+        private async Task<DbDataReader> GetLastRowAsync()
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var cmd = connection.CreateSelectCommand("SELECT * FROM T WHERE K=@k",
+                    new SpannerParameterCollection {{"K", _lastKey, SpannerDbType.String}});
```

Comment:
I matched the java test schema verbatim maybe for better (or worse).  But if its only a nit, I'd prefer to keep exactly in sync (including case).